pkg/segment/writer/startree_test.go

```diff
@@ -769,6 +770,61 @@ func getTotalLevels(node *Node) int {
 	return 1 + max
 }
 
+func (builder *StarTreeBuilder) DeriveEnc(wip WipBlock, recGroupByValues [][]string) (map[string]map[string]uint32, error) {
+	numRecs := wip.blockSummary.RecCount
+	mp := make(map[string]map[string]uint32)
+	for colNum := range builder.groupByKeys {
+		mp[builder.groupByKeys[colNum]] = make(map[string]uint32)
+	}
+
+	for recNum := uint16(0); recNum < numRecs; recNum += 1 {
+		for i, grpValues := range recGroupByValues[recNum] {
+			mp[builder.groupByKeys[i]][grpValues] = builder.wipRecNumToColEnc[i][recNum]
+		}
+	}
+
+	return mp, nil
+}
+
+func getTree(t *testing.T, root *Node, ans map[string][]utils.CValueEnclosure, key string) {
```

Comment:
nit: rename `ans`; I'm not sure what it means

Reply:
Actually I think this is the `answer`—the result you want from running `getTree`, right? In that case, maybe renaming `getTree` would be good to clarify what exactly we are getting

Reply:
addressed in the latest commit.